src/common/snippets/src/lowered/linear_ir.cpp

```diff
@@ -432,10 +432,25 @@ LinearIR::exprIt LinearIR::replace_with_expr(const std::vector<ExpressionPtr>& o
     const auto input_ports = new_expr_it->get()->get_input_ports();
     const auto output_ports = new_expr_it->get()->get_output_ports();
     for (const auto& old_expr : old_exprs) {
-        for (size_t i = 0; i < old_expr->get_input_count(); ++i)
-            m_loop_manager->replace_loop_ports(loop_ids, old_expr->get_input_port(i), input_ports);
-        for (size_t i = 0; i < old_expr->get_input_count(); ++i)
+        for (size_t i = 0; i < old_expr->get_input_count(); ++i) {
+            for (const auto& loop_id : loop_ids) {
+                const auto& loop_info = m_loop_manager->get_loop_info(loop_id);
+                if (!loop_info->is_loop_port(old_expr->get_input_port(i))) {
+                    continue;
+                }
+                for (size_t j = 0; j < new_expr_it->get()->get_input_count(); ++j) {
+                    if (new_expr_it->get()->get_input_port_connector(j)->get_source() ==
+                        old_expr->get_input_port_connector(i)->get_source()) {
+                        m_loop_manager->replace_loop_ports(loop_ids,
+                                                           old_expr->get_input_port(i),
+                                                           {new_expr_it->get()->get_input_port(j)});
+                    }
+                }
```

Comment:
If several inputs of `new_expr` are connected to the same `old_expr->get_input_port_connector(i)->get_source()`, only first input of `new_expr` will be added as `LoopPort`. Other inputs won't be added.

I suggest to select all needed inputs of `new_expr` at first and insert them to vector. And then we need to call `m_loop_manager->replace_loop_ports(loop_ids, old_expr->get_input_port(i), vector)` only once

What do you think?

Reply:
I agree, done